lib/origen_testers/smartest_based_tester/base.rb

```diff
@@ -338,9 +338,18 @@ def pattern_header(options = {})
         }.merge(options)
         pin_list = ordered_pins.map do |p|
           if Origen.app.pin_pattern_order.include?(p.id)
-            p.id.to_s.upcase # specified name overrides pin name
+            # specified name overrides pin name
+            if p.is_a?(Origen::Pins::PinCollection) || p.id != p.name
```

Comment:
Can we add a size > 1 qualifier to the pin collection case?

Not sure it would ever be hit in reality, but conceptually a single pin could find its way in here with a pin collection wrapper around it.